fury/data/fetcher.py

```diff
@@ -75,41 +75,47 @@ def _already_there_msg(folder):
     print(msg)
 
 
-def _get_file_md5(filename):
-    """Compute the md5 checksum of a file."""
-    md5_data = md5()
-    with open(filename, 'rb') as f:
-        for chunk in iter(lambda: f.read(128 * md5_data.block_size), b''):
-            md5_data.update(chunk)
-    return md5_data.hexdigest()
+def _get_file_sha(filename):
+    """Generates SHA checksum for the entire file in blocks of 256
 
+    Parameters
+    ----------
+    filename: str
+        The path to the file whose sha checksum is to be generated
 
-def check_md5(filename, stored_md5=None):
-    """Compute the md5 of filename.
+    """
+    sha256_data = sha256()
+    with open(filename, 'rb') as f:
+        for chunk in iter(lambda: f.read(256*sha256_data.block_size), b''):
+            sha256_data.update(chunk)
+    return sha256_data.hexdigest()
 
-    check if it matches with the supplied string md5
+def check_sha(filename, stored_sha256=None):
+    """Checks the generated sha checksum.
 
     Parameters
     ----------
-    filename : string
-        Path to a file.
-    md5 : string
-        Known md5 of filename to check against. If None (default), checking is
-        skipped
+    filename: str
+        The path to the file whose checksum is to be compared
+    stored_sha: string input, optional
+        Used to verify the generated SHA checksum
 
     """
-    if stored_md5 is not None:
-        computed_md5 = _get_file_md5(filename)
-        if stored_md5 != computed_md5:
-            msg = """The downloaded file, %s, does not have the expected md5
-   checksum of "%s". Instead, the md5 checksum was: "%s". This could mean that
-   something is wrong with the file or that the upstream file has been updated.
-   You can try downloading the file again or updating to the newest version of
-   Fury.""" % (filename, stored_md5,
-               computed_md5)
+    if stored_sha256 is not None:
+        computed_sha256 = _get_file_sha(filename).upper()
```

Comment:
I wonder why are you using `.upper()` ? Can you explain please?

Reply:
hashlib sha generates lowercase hash string, while on windows, the generated SHA hash was uppercase. As a result, i had to use upper to match both strings and proceed.

Reply:
ok, Thanks for this info!